tests/unit/amo/reducers/test_collections.js

```diff
@@ -495,6 +497,22 @@ describe(__filename, () => {
       expect(state.hasAddonBeenAdded).toEqual(false);
     });
 
+    it('sets a hasAddonBeenRemoved flag when beginning to remove an add-on from a collection', () => {
+      const state = reducer(
+        undefined,
+        removeAddonFromCollection({
+          addonId: 1,
+          collectionId: 3,
+          errorHandlerId: 'error-handler',
+          filters: {},
+          slug: 'some-collection',
+          username: 'some-user',
+        }),
+      );
+
+      expect(state.hasAddonBeenRemoved).toEqual(false);
```

Comment:
Since this is the same as what's in `initialState`, it would be better to apply `addonRemovedFromCollection()` to state before applying `removeAddonFromCollection()`.